frontend/src/components/BookmarkTab.tsx

```diff
@@ -23,18 +24,22 @@ const BookmarkTab = ({ bookmarks }: BookmarkTabProps) => {
             />
           ))}
       </div>
-      {bookmarks.length === 0 && (
-        <div className="flex justify-center mt-10">
-          <h6 className="text-xl font-bold py-2">
-            No bookmarks yet! You can bookmark blogs in the blogs page{' '}
-            <Link to="/signin" className="underline">
-              here
-            </Link>
-          </h6>
-        </div>
-      )}
+      {(bookmarks.length === 0 && user) ? (<div className="flex justify-center mt-10">
+        <h6 className="text-xl font-bold py-2">
+          No bookmarks yet! You can bookmark blogs in the blogs page{' '}
+          <Link to="/blogs" className="underline">
+            here
+          </Link>
+        </h6>
+      </div>) : (<div className="flex justify-center mt-10">
+        <h6 className="text-xl font-bold py-2">
+          No bookmarks yet! You can bookmark blogs in the blogs page{' '}
+          <Link to="/signin" className="underline">
+            here
+          </Link>
```

Comment:
Instead of repeating this code snippet twice in if else
```
<div className="flex justify-center mt-10">
        <h6 className="text-xl font-bold py-2">
          No bookmarks yet! You can bookmark blogs in the blogs page{' '}
          <Link to="/blogs" className="underline">
            here
          </Link>
        </h6>
      </div>
```
We can simply put the condition within like this

```bash
 <Link to={user? "/blogs" : "/signin"} ... />
 ```

Reply:
> Instead of repeating this code snippet twice in if else
> 
> ```
> <div className="flex justify-center mt-10">
>         <h6 className="text-xl font-bold py-2">
>           No bookmarks yet! You can bookmark blogs in the blogs page{' '}
>           <Link to="/blogs" className="underline">
>             here
>           </Link>
>         </h6>
>       </div>
> ```
> 
> We can simply put the condition within Link like this
> 
> ```shell
>  <Link to={user? "/blogs" : "/signin"} ... />
> ```

yes!